pyazhpc/azhpc.py

```diff
@@ -542,73 +551,62 @@ def do_slurm_resume(args):
 
     log.info(f"slurm resume for {args.nodes}")
     # first get the resource name
-    resource_name, brackets = re.search(r'([^[]*)\[?([\d\-\,]*)\]?', args.nodes).groups(0)
-    resource_list = []
-    if bool(brackets):
-        for part in brackets.split(","):
-            if "-" in part:
-                lo, hi = part.split("-")
-                assert len(lo) == 4, "expecting number width of 4"
-                assert len(hi) == 4, "expecting number width of 4"
-                for i in range(int(lo), int(hi) + 1):
-                    resource_list.append(f"{resource_name}{i:04d}")
-            else:
-                assert len(part) == 4, "expecting number width of 4"
-                resource_list.append(f"{resource_name}{part}")
-    else:
-        resource_list.append(resource_name)
-        resource_name = resource_name[:-4]
-    
-    template_resource = config.get("resources", {}).get(resource_name)
-    if not template_resource:
-        log.error(f"{resource_name} resource not found in config")
-        sys.exit(1)
-    if template_resource.get("type") != "slurm_partition":
-        log.error(f"invalid resource type for scaling")
-    
-    template_resource["type"] = "vm"
-    del template_resource["instances"]
-
-    log.info(f"resource_name= {resource_name}")
-    log.info("resource_list= " + ",".join(resource_list))
-    
-    config["resources"] = {}
-    for rname in resource_list:
-        config["resources"][rname] = template_resource
+    resource_names, resource_list = _nodelist_expand(args.nodes)
 
-    tpl = arm.ArmTemplate()
-    tpl.read_resources(config, False)
+    # Loop over all resources
+    for resource in resource_names:
```

Comment:
We are going to end up doing multiple deployments here (one per resource type).  It would be better to create a single config with *all* the instances.